src/algebra/ordered_monoid_lemmas.lean

```diff
@@ -84,10 +91,225 @@ for all `m ∈ M` and all elements `n₁, n₂ ∈ N`, if the relation `r` holds
+
+variables {M N}
+section has_mul
+variables [has_mul N]
+
+section has_le
+variables [has_le N]
+
+@[simp, to_additive]
+lemma mul_le_mul_iff_left [covariant_class N N (*) (≤)] [contravariant_class N N (*) (≤)]
+  (a : N) {b c : N} :
+  a * b ≤ a * c ↔ b ≤ c :=
+rel_iff_cov N N (*) (≤) a
+
+@[simp, to_additive]
+lemma mul_le_mul_iff_right
+  [covariant_class N N (function.swap (*)) (≤)] [contravariant_class N N (function.swap (*)) (≤)]
+  (a : N) {b c : N} :
+  b * a ≤ c * a ↔ b ≤ c :=
+rel_iff_cov N N (function.swap (*)) (≤) a
+
+@[to_additive]
+lemma mul_le_mul_left_n [covariant_class N N (*) (≤)]
+  (a : N) {b c : N} (bc : b ≤ c) :
+  a * b ≤ a * c :=
+covariant_class.covc _ bc
+
+@[to_additive]
+lemma le_of_mul_le_mul_left_n [contravariant_class N N (*) (≤)]
+  (a : N) {b c : N} (bc : a * b ≤ a * c) :
+  b ≤ c :=
+contravariant_class.covtc _ bc
+
+@[to_additive]
+lemma mul_le_mul_right_n [covariant_class N N (function.swap (*)) (≤)]
+  (a : N) {b c : N} (bc : b ≤ c) :
+  b * a ≤ c * a :=
+covariant_class.covc a bc
+
+@[to_additive]
+lemma le_of_mul_le_mul_right_n [contravariant_class N N (function.swap (*)) (≤)]
+  (a : N) {b c : N} (bc : b * a ≤ c * a) :
+  b ≤ c :=
+contravariant_class.covtc a bc
+
+/- The prime on this lemma is present only on the multiplicative version.  The unprimed version
+is taken by the analogous lemma for semiring, with an extra non-negativity assumption. -/
+@[to_additive add_le_add_left]
+lemma mul_le_mul_left' [covariant_class N N (*) (≤)] {b c : N} (bc : b ≤ c) (a : N) :
+  a * b ≤ a * c :=
+covariant_class.covc _ bc
+
+/- The prime on this lemma is present only on the multiplicative version.  The unprimed version
+is taken by the analogous lemma for semiring, with an extra non-negativity assumption. -/
+@[to_additive add_le_add_right]
+lemma mul_le_mul_right' [covariant_class N N (function.swap (*)) (≤)]
+  {b c : N} (bc : b ≤ c) (a : N) :
+  b * a ≤ c * a :=
+covariant_class.covc a bc
+
+end has_le
+
+section has_lt
+variables [has_lt N]
+
+@[simp, to_additive]
+lemma mul_lt_mul_iff_left [covariant_class N N (*) (<)] [contravariant_class N N (*) (<)]
+  (a : N) {b c : N} :
+  a * b < a * c ↔ b < c :=
+rel_iff_cov N N (*) (<) a
+
+@[simp, to_additive]
+lemma mul_lt_mul_iff_right
+  [covariant_class N N (function.swap (*)) (<)] [contravariant_class N N (function.swap (*)) (<)]
+  (a : N) {b c : N} :
+  b * a < c * a ↔ b < c :=
+rel_iff_cov N N (function.swap (*)) (<) a
+
+@[to_additive]
+lemma mul_lt_mul_left_n [covariant_class N N (*) (<)]
+  (a : N) {b c : N} (bc : b < c) :
+  a * b < a * c :=
+covariant_class.covc _ bc
+
+@[to_additive]
+lemma lt_of_mul_lt_mul_left_n [contravariant_class N N (*) (<)]
+  (a : N) {b c : N} (bc : a * b < a * c) :
+  b < c :=
+contravariant_class.covtc _ bc
+
+@[to_additive]
+lemma mul_lt_mul_right_n [covariant_class N N (function.swap (*)) (<)]
+  (a : N) {b c : N} (bc : b < c) :
+  b * a < c * a :=
+covariant_class.covc a bc
```

Comment:
What does the trailing `N` mean here?

Reply:
It was there to avoid clashing with the *o*ld names, as these are the *n*ew ones.  I am removing them now, but have to address the `linarith` issue, which is why they are still present!